tests/conftest.py

```diff
@@ -22,7 +22,7 @@
 
 
 @pytest.fixture(scope="session")
-def user_headers(session_mocker: MockerFixture):
+def user_headers(session_mocker: MockerFixture) -> dict[str, dict[str, str]]:
```

Comment:
Should we make this and dummy_users_data `typeddict`s? 🤔

Reply:
Could do. I tend to prefer dataclasses as they're slightly more rigid (and I think performant). Either way something for a separate PR